lib/speck.ex

```diff
@@ -103,6 +103,7 @@ defmodule Speck do
     end
   end
 
+  defp do_strict_validate(:any, value, _opts) when value != nil, do: value
```

Comment:
Does `:any` actually mean `!= nil`, or is it more like [Map.has_key](https://hexdocs.pm/elixir/1.17.3/Map.html#has_key?/2)? In other words, should the following be passed through, or is it really `:not_present`?

```ex
params = %{
  "param1" => nil
}
```

Reply:
I gave this some thought and this suggestion would change how other types/options behave. Lets observe how typed attribute currently behaves:

When a field is optional, we currently check if the `raw_value` is nil and if it isn't nil, then a best attempt coercion occurs.
When a field is strict and not optional, we check the value type and by extension validate it's not nil.

So if we unwrap that into steps, we have:
- type validation
- optional validation

Now, with the `:any` type, that is essentially the absence of `type` so if we remove type validation, everything else should remain the same. However, you did uncover something, the guard clause for `do_strict_validate(:any, value, _opts)` is redundant because it is already handled prior to the validation sub-routines. So, I can get rid of that.

Reply:
Result of our offline convo:

Move forward with the same behavior as the other types. `:any, optional: false` considers a `nil` value to be `:not_present`, in other words "any non-nil value". To let _everything_ pass through, use `:any, optional: true`.